libmachine/host/version.go

```diff
@@ -0,0 +1,44 @@
+package host
+
+import (
+	"fmt"
+
+	"github.com/docker/machine/libmachine/cert"
+	"github.com/samalba/dockerclient"
+)
+
+var DefaultDockerVersionner DockerVersionner = &BasicDockerVersionner{}
+
+type DockerVersionner interface {
+	DockerVersion(h *Host) (string, error)
+}
+
+type BasicDockerVersionner struct{}
+
+func (dv *BasicDockerVersionner) DockerVersion(host *Host) (string, error) {
+	client, err := dockerClient(host)
+	if err != nil {
+		return "", fmt.Errorf("Unable to query docker version: %s", err)
+	}
+
+	version, err := client.Version()
+	if err != nil {
+		return "", fmt.Errorf("Unable to query docker version: %s", err)
+	}
+
+	return version.Version, nil
+}
+
+func dockerClient(host *Host) (*dockerclient.DockerClient, error) {
```

Comment:
Why not move the code to get such a DockerClient struct to its own module, e.g. `mcndockerclient`? There are other parts of the code such as the provisioners which are likely to want to use import this as well.